contracts/SpokePool.sol

```diff
@@ -1,16 +1,176 @@
+     */
+    function _whitelistRoute(
+        address originToken,
+        address destinationToken,
+        bool isWethToken,
+        uint256 destinationChainId
+    ) internal {
+        // We use a correctly set destination token address as the determinant of whether a token path is whitelisted 
+        // or not. So, explicitly prevent the caller from unintentionally setting this to the zero address.
+        require(destinationToken != address(0), "Invalid spoke pool");
+        whitelistedDestinationRoutes[originToken][destinationChainId] = DestinationToken({
+            token: destinationToken,
+            isWethToken: isWethToken,
+            depositsEnabled: true
+        });
+
+        emit WhitelistRoute(originToken, destinationChainId, destinationToken, isWethToken);
+    }
+
+    /**
+     * @notice Enable/disable deposits for a whitelisted origin token.
+     */
+    function _setEnableDeposits(
+        address originToken, 
+        uint256 destinationChainId, 
+        bool depositsEnabled
+    ) internal onlyWhitelistedPath(originToken, destinationChainId) {
+        whitelistedDestinationRoutes[originToken][destinationChainId].depositsEnabled = depositsEnabled;
+        emit DepositsEnabled(originToken, destinationChainId, depositsEnabled);
+    }
+
+    /**************************************
+     *         DEPOSITOR FUNCTIONS        *
+     **************************************/
+
+    /**
+     * @notice Called by user to bridge funds from origin to destination chain.
+     * @dev The caller must first approve this contract to spend `amount` of `originToken`.
+     */
     function deposit(
         address originToken,
+        uint256 destinationChainId,
         uint256 amount,
         address recipient,
-        uint256 destinationChainId,
-        uint256 relayerFee
-    ) public {}
+        uint64 relayerFeePct,
+        uint64 quoteTimestamp
+    ) public payable onlyIfDepositsEnabled(originToken, destinationChainId) {
+        // We limit the relay fees to prevent the user spending all their funds on fees.
+        require(relayerFeePct <= 0.5e18, "invalid relayer fee");
+        // Note We assume that L2 timing cannot be compared accurately and consistently to L1 timing. Therefore, 
+        // `block.timestamp` is different from the L1 EVM's. Therefore, the quoteTimestamp must be within 10
+        // mins of the current time to allow for this variance.
+        // Note also that `quoteTimestamp` cannot be less than 10 minutes otherwise the following arithmetic can result
+        // in underflow. This isn't a problem as the deposit will revert, but the error might be unexpected for clients.
+        // Consider requiring `quoteTimestamp >= 10 minutes`.
+        require(
+            getCurrentTime() >= quoteTimestamp - 10 minutes && 
+            getCurrentTime() <= quoteTimestamp + 10 minutes &&
+            quoteTimestamp >= deploymentTime,
```

Comment:
Do we really need to check this? Wouldn't this check only be possible to violate in the first 10 minutes of the SpokePool's existence? Otherwise, you would always hit `getCurrentTime() <= quoteTimestamp + 10 minutes`, right? Also, what would happen if the quote timestamp were a few minutes before this contract is deployed (i.e. what is this trying to prevent?)? That wouldn't prevent the issue of the quote timestamp being before the HubPool whitelists a particular token, right?

Reply:
good point! would save a small bit of gas.

Reply:
Sure I can remove this check. 

Do you agree though that we should keep this deployment time stamp somewhere in the contract? We probably will need it in initiateRelay to block any deposit quote times before the deployment time